packages/delegation-process/test/createConsumerDelegation.test.ts

```diff
@@ -584,4 +607,115 @@ describe("create consumer delegation", () => {
       )
     );
   });
+
+  it("should throw an eserviceNotConsumerDelegable error if Eservice is not consumer delegable", async () => {
+    const delegatorId = generateId<TenantId>();
+    const authData = getRandomAuthData(delegatorId);
+    const delegator = {
+      ...getMockTenant(delegatorId),
+      externalId: {
+        origin: "IPA",
+        value: "test",
+      },
+    };
+
+    const delegate = {
+      ...getMockTenant(),
+      features: [
+        {
+          type: "DelegatedConsumer" as const,
+          availabilityTimestamp: new Date(),
+        },
+      ],
+    };
+
+    const eservice = {
+      ...getMockEService(generateId<EServiceId>(), delegatorId),
+      isDelegable: false,
+    };
+
+    await addOneTenant(delegator);
+    await addOneTenant(delegate);
+    await addOneEservice(eservice);
+
+    await expect(
+      delegationService.createConsumerDelegation(
+        {
+          delegateId: delegate.id,
+          eserviceId: eservice.id,
+        },
+        {
+          authData,
+          logger: genericLogger,
+          correlationId: generateId(),
+          serviceName: "DelegationServiceTest",
+        }
+      )
+    ).rejects.toThrowError(eserviceNotConsumerDelegable(eservice.id));
+  });
+
+  it("should throw a delegationRelatedAgreementExists error if an agreement exists", async () => {
+    const delegatorId = generateId<TenantId>();
+    const authData = getRandomAuthData(delegatorId);
+    const delegator = {
+      ...getMockTenant(delegatorId),
+      externalId: {
+        origin: "IPA",
+        value: "test",
+      },
+    };
+
+    const delegate = {
+      ...getMockTenant(),
+      features: [
+        {
+          type: "DelegatedConsumer" as const,
+          availabilityTimestamp: new Date(),
+        },
+      ],
+    };
+
+    const eservice = {
+      ...getMockEService(generateId<EServiceId>(), delegatorId),
+      isDelegable: true,
+    };
+
+    const randomAgreementState = randomArrayItem([
+      agreementState.active,
+      agreementState.pending,
+      agreementState.suspended,
+    ]);
```

Comment:
You could also do `it.each` to test each state as a dedicated test - no strong opinion on this